agent/engine/execcmd/manager_start_linux.go

```diff
@@ -135,35 +140,45 @@ func (m *manager) doStartAgent(ctx context.Context, client dockerapi.DockerClien
 	}
 	execRes, err := client.CreateContainerExec(ctx, containerId, execCfg, dockerclient.ContainerExecCreateTimeout)
 	if err != nil {
+		execMD.StoppedReason = fmt.Sprintf("create start error: %v", err)
+		container.SetKnownExecCommandAgentStatus(apicontainerstatus.ManagedAgentStopped)
 		return StartError{error: err, retryable: true}
 	}
 
 	seelog.Debugf("Task engine [%s]: created ExecCommandAgent for container: %s -> docker exec id: %s", task.Arn, containerId, execRes.ID)
 
 	err = client.StartContainerExec(ctx, execRes.ID, dockerclient.ContainerExecStartTimeout)
 	if err != nil {
+		execMD.StoppedReason = fmt.Sprintf("pre start error: %v", err)
+		container.SetKnownExecCommandAgentStatus(apicontainerstatus.ManagedAgentStopped)
 		return StartError{error: err, retryable: true}
 	}
 	seelog.Debugf("Task engine [%s]: sent ExecCommandAgent start signal for container: %s ->  docker exec id: %s", task.Arn, containerId, execRes.ID)
 
 	inspect, err := client.InspectContainerExec(ctx, execRes.ID, dockerclient.ContainerExecInspectTimeout)
 	if err != nil {
+		execMD.StoppedReason = fmt.Sprintf("inspect error: %v", err)
+		container.SetKnownExecCommandAgentStatus(apicontainerstatus.ManagedAgentStopped)
 		return StartError{error: err, retryable: true}
 	}
 	seelog.Debugf("Task engine [%s]: inspect ExecCommandAgent for container: %s -> pid: %d, exitCode: %d, running:%v, err:%v",
 		task.Arn, containerId, inspect.Pid, inspect.ExitCode, inspect.Running, err)
 
 	if !inspect.Running { //TODO: [ecs-exec] retry only for certain exit codes?
+		execMD.StoppedReason = fmt.Sprintf("post start error: Exit code: %d", inspect.ExitCode)
+		container.SetKnownExecCommandAgentStatus(apicontainerstatus.ManagedAgentStopped)
 		return StartError{
 			error:     fmt.Errorf("ExecCommandAgent could not start for container (exitCode=%d)", inspect.ExitCode),
 			retryable: true,
 		}
 	}
 	seelog.Infof("Task engine [%s]: started ExecCommandAgent for container: %s ->  docker exec id: %s", task.Arn, containerId, execRes.ID)
-	container.SetExecCommandAgentMetadata(&apicontainer.ExecCommandAgentMetadata{
-		PID:          strconv.Itoa(inspect.Pid),
-		DockerExecID: execRes.ID,
-		CMD:          execAgentCmd,
-	})
+	nowTime := time.Now()
+	execMD.PID = strconv.Itoa(inspect.Pid)
+	execMD.DockerExecID = execRes.ID
+	execMD.CMD = execAgentCmd
+	execMD.StartedAt = &nowTime
+	container.SetExecCommandAgentMetadata(execMD)
+	container.SetKnownExecCommandAgentStatus(apicontainerstatus.ManagedAgentRunning)
```

Comment:
As mentioned before, it's probably better if we do this one level up in `m.StartAgent`